src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/util/BulkConnectionRetryWrapper.java

```diff
@@ -0,0 +1,100 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.plugin.source.batch.util;
+
+import com.sforce.async.AsyncApiException;
+import com.sforce.async.BatchInfo;
+import com.sforce.async.BatchInfoList;
+import com.sforce.async.BulkConnection;
+import com.sforce.async.JobInfo;
+import dev.failsafe.Failsafe;
+import dev.failsafe.RetryPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStream;
+import java.time.Duration;
+
+/**
+ * BulkConnectionRetryWrapper class to retry all the salesforce api calls in case of failure.
+ */
+public class BulkConnectionRetryWrapper {
+
+  private final BulkConnection bulkConnection;
+  private final RetryPolicy retryPolicy;
+  private static final Logger LOG = LoggerFactory.getLogger(BulkConnectionRetryWrapper.class);
+
+  public BulkConnectionRetryWrapper(BulkConnection bulkConnection) {
+    this.bulkConnection = bulkConnection;
+    this.retryPolicy = getRetryPolicy(5L, 80L, 5);
+  }
+
+  public JobInfo createJob(JobInfo jobInfo) {
+    Object resultJobInfo = Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while creating job"))
+      .get(() -> bulkConnection.createJob(jobInfo));
+    return (JobInfo) resultJobInfo;
+  }
+
+  public JobInfo getJobStatus(String jobId) {
+    Object resultJobInfo = Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while getting job status"))
+      .get(() -> bulkConnection.getJobStatus(jobId));
+    return (JobInfo) resultJobInfo;
+  }
+
+  public void updateJob(JobInfo jobInfo) {
+    Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while updating job."))
+      .get(() -> bulkConnection.updateJob(jobInfo));
+  }
+
+  public BatchInfoList getBatchInfoList(String jobId) throws AsyncApiException {
+    Object batchInfoList = Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while getting batch info list"))
+      .get(() -> bulkConnection.getBatchInfoList(jobId));
+    return (BatchInfoList) batchInfoList;
+  }
+
+  public BatchInfo getBatchInfo(String jobId, String batchId) throws AsyncApiException {
+    Object batchInfo = Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while getting batc status"))
+      .get(() -> bulkConnection.getBatchInfo(jobId, batchId));
+    return (BatchInfo) batchInfo;
+  }
+
+  public InputStream getBatchResultStream(String jobId, String batchId) throws AsyncApiException {
+    Object inputStream = Failsafe.with(retryPolicy)
+      .onFailure(event -> LOG.info("Failed while getting batch result stream"))
+      .get(() -> bulkConnection.getBatchResultStream(jobId, batchId));
+    return (InputStream) inputStream;
+  }
+
+  public static RetryPolicy<Object> getRetryPolicy(Long initialRetryDuration, Long maxRetryDuration,
```

Comment:
Use the one from SplitUtil class and call it once while creating retry wrapper. If retryDisabled, handle it in this class constructor.